From 1fdd24e6ff37ebce16a324b72c01351dccd2df91 Mon Sep 17 00:00:00 2001
From: nift4 <nift4@protonmail.com>
Date: Wed, 23 Aug 2023 13:04:44 +0200
Subject: [PATCH] LatinIME: recreate keyboard theme when configuration changes

* checking if resources object is equal is not enough
* fixes Material You accent not updating when switching light/dark mode

Change-Id: I798041cdcbf6b7bbcdc69f4edbb9d0bf392bbb69
---
 .../com/android/inputmethod/keyboard/KeyboardSwitcher.java | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/java/src/com/android/inputmethod/keyboard/KeyboardSwitcher.java b/java/src/com/android/inputmethod/keyboard/KeyboardSwitcher.java
index 0e813ae5c..de6a81272 100644
--- a/java/src/com/android/inputmethod/keyboard/KeyboardSwitcher.java
+++ b/java/src/com/android/inputmethod/keyboard/KeyboardSwitcher.java
@@ -69,6 +69,9 @@ public final class KeyboardSwitcher implements KeyboardState.SwitchActions {
     private KeyboardTheme mKeyboardTheme;
     private Context mThemeContext;
     private int mActiveUiMode;
+    // Cache the configuration to detect changes, Context/Resources changes it in real time,
+    // we need to keep the one used for creation to detect changes
+    private Configuration mThemeConfig;
 
     private static final KeyboardSwitcher sInstance = new KeyboardSwitcher();
 
@@ -107,9 +110,11 @@ public final class KeyboardSwitcher implements KeyboardState.SwitchActions {
                 != (context.getResources().getConfiguration().uiMode
                     & Configuration.UI_MODE_NIGHT_MASK);
         if (mThemeContext == null || !keyboardTheme.equals(mKeyboardTheme) || darkModeChanged
-                || !mThemeContext.getResources().equals(context.getResources())) {
+                || !mThemeContext.getResources().equals(context.getResources())
+                || !mThemeConfig.equals(context.getResources().getConfiguration())) {
             mKeyboardTheme = keyboardTheme;
             mThemeContext = new ContextThemeWrapper(context, keyboardTheme.mStyleId);
+            mThemeConfig = mThemeContext.getResources().getConfiguration();
             mActiveUiMode = context.getResources().getConfiguration().uiMode;
             KeyboardLayoutSet.onKeyboardThemeChanged();
             return true;
-- 
2.49.0

